tests/selenium/test_sizing.py

```diff
@@ -0,0 +1,270 @@
+        assert abs(target_cell.size["width"] - table_r1c1_cells[i].size["width"]) <= 1
+
+    if len(table_r0c0_cells) != 0:
+        assert abs(target_cell.size["width"] - table_r0c0_cells[i].size["width"]) <= 1
+
+    if len(table_r0c1_cells) != 0:
+        assert abs(target_cell.size["width"] - table_r0c1_cells[i].size["width"]) <= 1
+
+    if len(table_r1c0_cells) != 0:
+        assert abs(target_cell.size["width"] - table_r1c0_cells[i].size["width"]) <= 1
+
+
+def test_szng001_widths_on_style_change(test):
+    base_props = dict(
+        data=[
+            {"a": 85, "b": 601, "c": 891},
+            {"a": 967, "b": 189, "c": 514},
+            {"a": 398, "b": 262, "c": 743},
+            {"a": 89, "b": 560, "c": 582},
+            {"a": 809, "b": 591, "c": 511},
+        ],
+        columns=[
+            {"id": "a", "name": "A"},
+            {"id": "b", "name": "B"},
+            {"id": "c", "name": "C"},
+        ],
+        style_data={
+            "width": 100,
+            "minWidth": 100,
+            "maxWidth": 100,
+            "border": "1px solid blue",
+        },
+        row_selectable="single",
+        row_deletable=True,
+    )
+
+    styles = [
+        dict(
+            style_table=dict(
+                width=500, minWidth=500, maxWidth=500, paddingBottom=10, display="none"
+            )
+        ),
+    ]
+
+    fixes = [
+        dict(),
+        dict(fixed_columns=dict(headers=True)),
+        dict(fixed_rows=dict(headers=True)),
+        dict(fixed_columns=dict(headers=True), fixed_rows=dict(headers=True)),
+        dict(fixed_columns=dict(headers=True, data=1)),
+        dict(fixed_rows=dict(headers=True, data=1)),
+        dict(
+            fixed_columns=dict(headers=True, data=1),
+            fixed_rows=dict(headers=True, data=1),
+        ),
+    ]
+
+    variations = []
+    i = 0
+    for style in styles:
+        for fix in fixes:
+            variations.append({**style, **fix, **base_props, "id": "table{}".format(i)})
+            i = i + 1
+
+    variations_range = range(0, len(variations))
+
+    tables = [DataTable(**variation) for i, variation in enumerate(variations)]
+
+    app = dash.Dash(__name__)
+    app.layout = Div(
+        children=[
+            Button(id="btn", children="Click me"),
+            DataTable(
+                **base_props,
+                id="table350",
+                style_table=dict(
+                    width=350, minWidth=350, maxWidth=350, paddingBottom=10
+                )
+            ),
+            DataTable(
+                **base_props,
+                id="table500",
+                style_table=dict(
+                    width=500, minWidth=500, maxWidth=500, paddingBottom=10
+                )
+            ),
+            DataTable(
+                **base_props,
+                id="table750",
+                style_table=dict(
+                    width=750, minWidth=750, maxWidth=750, paddingBottom=10
+                )
+            ),
```

Comment:
These tables are used to make sure the test tables are behaving as expected when updated through the callback below